libglusterfs/src/event-epoll.c

```diff
@@ -78,22 +78,16 @@ __event_slot_alloc(struct event_pool *event_pool, int fd,
 
 retry:
 
-    while (i < EVENT_EPOLL_TABLES) {
+    if (i < EVENT_EPOLL_TABLES) {
```

Comment:
nit (and perhaps not related to this patch) - can we just use table_idx and get rid of i? Why do we need both?

Reply:
This condition is always true and should be removed (`i` will always be `< EVENT_EPOLL_TABLES` unless it's defined as 0, but it shouldn't).

The other question is why we have `EVENT_EPOLL_TABLES` defined as 1024 if we only use the first one ?

Maybe the problem is not the dead code but that in some case we should keep iterating the remaining tables ?

With `EVENT_EPOLL_SLOTS` also defined as 1024, it means that we may only have 1024 active epoll entries, which translates into a limit of ~1000 connections (1 epoll slot per connection). For big volumes, or for servers with many bricks using brick-mux, that could be a problem.

Reply:
> This condition is always true and should be removed (`i` will always be `< EVENT_EPOLL_TABLES` unless it's defined as 0, but it shouldn't).
> 
> The other question is why we have `EVENT_EPOLL_TABLES` defined as 1024 if we only use the first one ?

I've asked this @ https://github.com/gluster/glusterfs/issues/3228

> 
> Maybe the problem is not the dead code but that in some case we should keep iterating the remaining tables ?
> 
> With `EVENT_EPOLL_SLOTS` also defined as 1024, it means that we may only have 1024 active epoll entries, which translates into a limit of ~1000 connections (1 epoll slot per connection). For big volumes, or for servers with many bricks using brick-mux, that could be a problem.